rmf_fleet_adapter/src/rmf_fleet_adapter/agv/FleetUpdateHandle.cpp

```diff
@@ -1032,14 +1035,24 @@ void FleetUpdateHandle::add_robot(
           fleet->_pimpl->default_maximum_delay,
           state,
           fleet->_pimpl->task_planner
-        });
+        }
+      );
 
       // We schedule the following operations on the worker to make sure we do not
       // have a multiple read/write race condition on the FleetUpdateHandle.
       worker.schedule(
-        [context, fleet, node = fleet->_pimpl->node,
+        [fleet_wptr = std::weak_ptr<FleetUpdateHandle>(fleet), 
+        node_wptr = std::weak_ptr<Node>(fleet->_pimpl->node),
+        context = std::move(context),
         handle_cb = std::move(handle_cb)](const auto&)
         {
+          auto fleet = fleet_wptr.lock();
+          if(!fleet) return;
```

Comment:
There are some style tweaks needed here, i.e. a space in `if (` and a newline before `return;`.

Reply:
Yea I'll run uncrustify as a batch process.